src/App.tsx

```diff
@@ -1,24 +1,158 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { Errors } from './components/Errors';
+import { TodoContent } from './components/TodoContent';
 import { UserWarning } from './UserWarning';
+import {
+  getTodos, addTodo, deleteTodo, patchTodo,
+} from './api/todos';
+import { Todo } from './types/Todo';
+import { Filter } from './types/Filter';
+import { ErrorMessages } from './types/ErrorMessages';
 
-const USER_ID = 0;
+const USER_ID = 6232;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filteredTodos, setFilteredTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState<ErrorMessages | null>(null);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isInputDisabled, setIsInputDisabled] = useState(false);
+  const [loadingAll, setLoadingAll] = useState(false);
+
+  const filterTodos = (filterBy: Filter) => {
+    setFilteredTodos(
+      todos.filter((todo) => {
+        switch (filterBy) {
+          case Filter.active:
+            return todo.completed === false;
+
+          case Filter.completed:
+            return todo.completed === true;
+
+          default:
+            return todo;
+        }
+      }),
+    );
+  };
+
+  const createTodo = async (title: string) => {
+    setIsInputDisabled(true);
+
+    const newTodo = {
+      title,
+      userId: USER_ID,
+      completed: false,
+    };
+
+    setTempTodo({ ...newTodo, id: 0 });
+    await addTodo(newTodo)
+      .then((response) => {
+        setTempTodo(null);
+
+        setFilteredTodos((state) => [...state, response]);
+      })
+      .catch(() => {
+        setError(ErrorMessages.addTodo);
+      });
+
+    setIsInputDisabled(false);
+  };
+
+  const removeTodo = async (id: number) => {
+    await deleteTodo(id)
+      .then(() => {
+        setFilteredTodos(filteredTodos.filter((todo) => todo.id !== id));
+      })
+      .catch(() => {
+        setError(ErrorMessages.deleteTodo);
+      });
+  };
+
+  const updateTodo = async (todo: Todo, update: 'title' | 'complete') => {
+    const data
+      = update === 'title' ? todo : { ...todo, completed: !todo.completed };
+
+    await patchTodo(data)
+      .then((res) => {
+        setFilteredTodos(
+          filteredTodos.map((t) => {
```

Comment:
```suggestion
          (filteredTodos) => filteredTodos.map((t) => {
```
I would recommend you to use the current filteredTodos value instead of reading the variable again.